src/selectors/mappedBreakpoints.js

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+import { makeLocationId } from "../utils/breakpoint";
+
+import {
+  getSources,
+  getSourceInSources,
+  getBreakpoints,
+  getPauseReason,
+  getSelectedSource,
+  getTopFrame
+} from "../selectors";
+
+import { isGeneratedId } from "devtools-source-map";
+
+import { createSelector } from "reselect";
+
+function formatBreakpoint(
+  sources,
+  selectedSource,
+  breakpoint
+): LocalBreakpoint {
+  if (!selectedSource) {
+    return;
+  }
+
+  let location = breakpoint.location;
+  let text = breakpoint.originalText;
+  const condition = breakpoint.condition;
+  const locationId = makeLocationId(location);
+  const source = getSourceInSources(sources, location.sourceId);
+
+  if (isGeneratedId(selectedSource.id)) {
+    location = breakpoint.generatedLocation || breakpoint.location;
+    text = breakpoint.text;
+  }
+  const localBP = { locationId, location, text, source, condition };
+
+  return localBP;
+}
+
+function _getMappedBreakpoints(breakpoints, sources, selectedSource) {
+  return breakpoints
```

Comment:
maybe check for the selectedSource here so that we can be extra cautious

Reply:
Returning a `I.Map()` seems like the best idea, IMO, because it can prevent breakage elsewhere.